presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -160,137 +162,139 @@ public void setCreateTableAsSelectNoOp(boolean createTableAsSelectNoOp)
 
     public void setAggregates(QuerySpecification node, List<FunctionCall> aggregates)
     {
-        this.aggregates.put(node, aggregates);
+        this.aggregates.put(NodeRef.of(node), aggregates);
     }
 
     public List<FunctionCall> getAggregates(QuerySpecification query)
     {
-        return aggregates.get(query);
+        return aggregates.get(NodeRef.of(query));
     }
 
     public void setOrderByAggregates(OrderBy node, List<Expression> aggregates)
     {
-        this.orderByAggregates.put(node, ImmutableList.copyOf(aggregates));
+        this.orderByAggregates.put(NodeRef.of(node), ImmutableList.copyOf(aggregates));
     }
 
-    public List<Expression> getOrderByAggregates(OrderBy query)
+    public List<Expression> getOrderByAggregates(OrderBy node)
     {
-        return orderByAggregates.get(query);
+        return orderByAggregates.get(NodeRef.of(node));
     }
 
     public IdentityLinkedHashMap<Expression, Type> getTypes()
     {
-        return new IdentityLinkedHashMap<>(types);
+        return NodeRefCollections.toIdentityMap(types);
```

Comment:
@sopel39 if you looked how `IdentityLinkedHashMap` was implemented, you'd see that it is very not straightforward to (so to say) to implement view of the map with different type.

Since those linear-cost transformations are _temporary_ (they all disappear before end of the PR), I decided I do not care about performance of the _interim_ commits. I _do_ care about performance of the eventual code, but it is not degraded in any way